tools/contrib/subset_data.py

```diff
@@ -0,0 +1,1096 @@
+        print( "----------------------------------------------------------------------")
+        print("Creating DATM domain file at ", self.plon, self.plat, ".")
+        # create 1d coordinate variables to enable sel() method
+        filename = self.fdatmdomain_in
+        f2 = self.create_1d_coord(filename, 'xc','yc','ni','nj')
+        # extract gridcell closest to plon/plat
+        f3 = f2.sel(ni=self.plon,nj=self.plat,method='nearest')
+        # expand dimensions
+        f3 = f3.expand_dims(['nj','ni'])
+        wfile=self.fdatmdomain_out
+        #update attributes
+        self.update_metadata(f3)
+        f3.attrs['Created_from'] = self.fdatmdomain_in 
+        # mode 'w' overwrites file
+        f3.to_netcdf(path=wfile, mode='w')
+        print('Successfully created file (fdatmdomain_out) :'+self.fdatmdomain_out)
+        f2.close(); f3.close()
+
+    def extract_datm_at(self, file_in, file_out):
+        # create 1d coordinate variables to enable sel() method
+        f2 = self.create_1d_coord(file_in, 'LONGXY','LATIXY','lon','lat')
+        # extract gridcell closest to plon/plat
+        f3  = f2.sel(lon=self.plon,lat=self.plat,method='nearest')
+        # expand dimensions
+        f3 = f3.expand_dims(['lat','lon'])
+        # specify dimension order 
+        f3 = f3.transpose(u'scalar','time','lat','lon')
+
+        #update attributes
+        self.update_metadata(f3)
+        f3.attrs['Created_from'] = file_in 
+        # mode 'w' overwrites file
+        f3.to_netcdf(path=file_out, mode='w')
+        print('Successfully created file :'+ file_out)
+        f2.close(); f3.close()
+
+    def create_datm_at_point(self):
+        print( "----------------------------------------------------------------------")
+        print("Creating DATM files at ", self.plon, self.plat, ".")
+        #--  specify subdirectory names and filename prefixes
+        solrdir = 'Solar/'
+        precdir = 'Precip/'
+        tpqwldir = 'TPHWL/'
+        prectag = 'clmforc.GSWP3.c2011.0.5x0.5.Prec.'
+        solrtag = 'clmforc.GSWP3.c2011.0.5x0.5.Solr.'
+        tpqwtag = 'clmforc.GSWP3.c2011.0.5x0.5.TPQWL.'
+
+        #--  create data files  
+        infile=[]
+        outfile=[]
+        for y in range(self.datm_syr,self.datm_eyr+1):
+          ystr=str(y)
+          for m in range(1,13):
+             mstr=str(m) 
+             if m < 10:
+                mstr='0'+mstr
+
+             dtag=ystr+'-'+mstr
+
+             fsolar=self.dir_input_datm+solrdir+solrtag+dtag+'.nc'
+             fsolar2=self.dir_output_datm+solrtag+self.tag+'.'+dtag+'.nc'
+             fprecip=self.dir_input_datm+precdir+prectag+dtag+'.nc'
+             fprecip2=self.dir_output_datm+prectag+self.tag+'.'+dtag+'.nc'
+             ftpqw=self.dir_input_datm+tpqwldir+tpqwtag+dtag+'.nc'
+             ftpqw2=self.dir_output_datm+tpqwtag+self.tag+'.'+dtag+'.nc'
+
+             infile+=[fsolar,fprecip,ftpqw]
+             outfile+=[fsolar2,fprecip2,ftpqw2]
```

Comment:
It's important to have different files if they are on different time intervals or different time handling. For the single tower site cases where everything is on the same time interval (for example hourly) you don't need three files. For the standard global cases we handle time differently on the three files. The solar data uses cosine solar zenith angle scaling, the precipitation uses the nearest, and the other one uses a linear time interpolation.

Reply:
I'm on board with this, @ekluzek I wonder how we can make this easier to set up out of the box for these two cases so that users are more likely to get it right, especially re. the `user_nl_datm` modifications that need to be set for the standard global case that should NOT be set in a tower site case.

Is this easiest to handle by setting a flag in #1278 that identifies `SinglePointGlobal` vs. `SinglePointTower` for these two ways of handling the datm file structure, interpolation, etc?  Alternatively we could call these `PtFromGlobal` and `PtFromTower`.

Reply:
Does this script find the nearest point for forcing data files as well as changing CTSM input files? If so yes you need both options. I don't think you really need a flag in the compset to define the two, since that's already done inside the compset. 1PT compsets assume the tower format and GSWP3 assumes the global 3-file format.